### PR TITLE
Remove Ubuntu Trusty

### PR DESCRIPTION
**Pull request by blag**
This PR removes all Ubuntu Trusty support.

I'm not entirely sure what I'm doing, so any help or direction is more than welcome.

* [x] Remove Trusty from CircleCI config

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -123,8 +123,9 @@ jobs:
       # The primary container is an instance of the first list image listed. Your build commands run in this container.
       - image: circleci/ruby:2.4
     working_directory: /tmp/deploy
+    parallelism: 5
```
Comment (armab) [inline .circleci/config.yml:126]:
```suggestion
```

Parallelism in deploying the packages is already handled below with `parallel`.

Comment (blag) [reply .circleci/config.yml]: Thanks, fixed!

**Comment by punkrokk** [comment]
FWIW, I am not nearly as knowledgeable on Ubuntu packaging as I am on EL. I can do a once over in this, but until I can commit more time to getting up to speed on the U/deb packaging, I may not be of much value.

**Comment by armab** [comment]
Alright, @blag let's please remove all `*.upstart` files first.

**Comment by armab** [comment]
@blag Any update on this PR?

**Comment by blag** [comment]
Thank you for getting this over the finish line! 😄

**Pull request merged**